widget/embedded/src/hooks/useSearchCustomTokens/useSearchCustomTokens.ts

```diff
@@ -0,0 +1,74 @@
+import type { Token } from 'rango-sdk';
+
+import { useCallback, useRef, useState } from 'react';
+
+import { httpService } from '../../services/httpService';
+import { useAppStore } from '../../store/AppStore';
+import { createAssetKey } from '../../store/utils/wallets';
+import { debounce } from '../../utils/common';
+
+import { DEBOUNCE_DELAY } from './useSearchCustomTokens.constants';
+
+export function useSearchCustomTokens() {
```

Comment:
Don't you think it would be better to add interfaces for result of  `useSearchCustomTokens`?
```suggestion

interface SearchCustomTokenHookResult {
  fetch: (query: string, blockchain?: string) => void;
  cancel: () => void;
  loading: boolean;
  tokens: Token[];
  error: string | null;
}
export function useSearchCustomTokens(): SearchCustomTokenHookResult {
```